### PR TITLE
[GPU] Coverity fixes

### PR DESCRIPTION
**Pull request by Lyamin-Roman**
### Tickets:
 - *108957*

### CODE REVIEW

File: src/plugins/intel_gpu/src/plugin/ops/eye.cpp [hunk 1]
```diff
@@ -21,14 +21,18 @@ static void CreateEyeOp(Program& p, const std::shared_ptr<ngraph::op::v9::Eye>&
...
 
     size_t dim_size = std::max(os_sz, static_cast<size_t>(4));
     InferenceEngine::SizeVector dims(dim_size, 1);
     for (size_t i = dim_size, j = os_sz; i > 0 && j > 0; --i, --j) {
         dims[i - 1] = output_shapes[j - 1];
     }
     const ngraph::op::v0::Constant* constant = dynamic_cast<const ngraph::op::v0::Constant*>(op->get_input_node_ptr(2));
+    if (!constant) {
+        OPENVINO_ASSERT("Unsupported parameter nodes type in ", op->get_friendly_name(), " (", op->get_type_name(), ")");
+    }
```
[the remark this answers is not in the document]
Comment (Lyamin-Roman) [reply src/plugins/intel_gpu/src/plugin/ops/eye.cpp]: Fixed

File: src/plugins/intel_gpu/src/graph/graph_optimizer/prepare_quantization.cpp [hunk 2]
```diff
@@ -713,9 +713,6 @@ void prepare_quantization::prepare_asymmetric_quantization(program &p, convoluti
 
     // Remove sub operations from the graph and set correct users for zero points and inputs
     if (asymmetric_data) {
-        if (!new_a_zp || !new_input)
-            CLDNN_ERROR_MESSAGE(new_conv_node.id(), "Unexpected nullptr in asymmetric quantization for activations optimization");
```
Comment (isanghao) [inline src/plugins/intel_gpu/src/graph/graph_optimizer/prepare_quantization.cpp:717]: why is it removed instead of changing to `OPENVINO_ASSERT`?

File: src/plugins/intel_gpu/src/plugin/ops/rnn.cpp [hunk 3]
```diff
@@ -140,7 +138,7 @@ static void CreateLSTMCellOp(Program& p, const std::shared_ptr<ngraph::op::v4::L
     cldnn::tensor FCInputReshapeSz = { lstm_batch_size, inputShape.spatial[0] + inStateShape.spatial[0], 1, 1 };
     p.add_primitive(*op, cldnn::reshape(FCInputReshapeID, cldnn::input_info(input_concatID), FCInputReshapeSz));
 
-    p.add_primitive(*op, cldnn::fully_connected(lstm_fc_id, cldnn::input_info(FCInputReshapeID), WRconcatID, hasBias ? bias.pid : ""));
+    p.add_primitive(*op, cldnn::fully_connected(lstm_fc_id, cldnn::input_info(FCInputReshapeID), WRconcatID, bias.pid));
```
Comment (isanghao) [inline src/plugins/intel_gpu/src/plugin/ops/rnn.cpp:141]: is this also coverity? This is quite a huge PR and if it doesn't belong to coverity fix, I think it would be better to put it in a separate PR.

File: src/plugins/intel_gpu/src/plugin/infer_request_legacy.cpp [hunk 4]
```diff
@@ -543,13 +543,11 @@ void InferRequestLegacy::SetGraph(std::shared_ptr<Graph> graph) {
...
 void InferRequestLegacy::SetBatch(int new_batch) {
     OV_ITT_SCOPED_TASK(itt::domains::intel_gpu_plugin, "InferRequestLegacy::SetBatch");
-    if (m_graph->GetMaxDynamicBatchSize() < 0)
-        IE_THROW() << "Dynamic batch is not enabled.";
 
-     if (new_batch < 1 || static_cast<size_t>(new_batch) > m_graph->GetMaxDynamicBatchSize()) {
-        IE_THROW() << "Invalid dynamic batch size " << new_batch <<
-            " for this request. Got: " << new_batch << ". Expected value in range [1;" << m_graph->GetMaxDynamicBatchSize() << "]";
-    }
+    OPENVINO_ASSERT(new_batch != 0 && static_cast<size_t>(new_batch) <= m_graph->GetMaxDynamicBatchSize(),
```
Comment (sshlyapn) [inline src/plugins/intel_gpu/src/plugin/infer_request_legacy.cpp:547]: As I can see, SetBatch has default new_batch=-1, is it your intention to cover this case with static_cast<size_t>(new_batch) casting?

File: src/plugins/intel_gpu/src/graph/graph_optimizer/prepare_quantization.cpp (hunk 2 above)
Comment (Lyamin-Roman) [reply src/plugins/intel_gpu/src/graph/graph_optimizer/prepare_quantization.cpp:717]: Because this will never be null in case of asymmetric_data

File: src/plugins/intel_gpu/src/plugin/ops/rnn.cpp (hunk 3 above)
Comment (Lyamin-Roman) [reply src/plugins/intel_gpu/src/plugin/ops/rnn.cpp:141]: Yes, immutable `hasBias` value leads to dead code

File: src/plugins/intel_gpu/src/plugin/infer_request_legacy.cpp (hunk 4 above)
[the remark this answers is not in the document]
Comment (Lyamin-Roman) [reply src/plugins/intel_gpu/src/plugin/infer_request_legacy.cpp]: Can be, fixed it

**Review by sshlyapn — APPROVED** [review]
